libs/core/src/main/java/org/opensearch/core/index/IndexingRetryException.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.core.index;
+
+import org.opensearch.OpenSearchException;
+
+/**
+ * This exception indicates that retry has been made during indexing for AppendOnly index. If the response of any
+ * indexing request contains this Exception in the response, we do not need to add a translog entry for this request.
+ *
+ * @opensearch.internal
+ */
+public class IndexingRetryException extends OpenSearchException {
```

Comment:
Can we make it explicit AppendOnlyIndexOperationRetryException

Reply:
Ack.